File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.go
func productExceptSelf(nums []int) []int {
    prev := 1
    prefix := make([]int, len(nums))
    for i := range nums {
        prefix[i] = prev
        prev = nums[i] * prev
    }
    prev = 1
    for i := len(nums)-1; i >= 0; i-- {
        prefix[i] *= prev
        curr := prev * nums[i]
        prev = curr
    }
    return prefix
}

File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]
        curr = nums[0]
        for i in range(1, len(nums)):
            num = nums[i]
            curr = max(curr+num, num)
            res = max(res, curr)
        return res